platform/accessibility/common-issues/focus-order-guidelines.md

```diff
@@ -0,0 +1,70 @@
+Applications must have elements that receive [focus in an order that preserves meaning and usability](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html) for sighted, keyboard, and assistive technology users. All users should be able to determine where their keyboard focus is quickly, and understand when that focus changes based on changes in the user interface.
+
+## Acceptance Criteria
+
+- [ ] As a visual, keyboard, or screen reader user, I want to understand the focus order and navigate your web application successfully.
+- [ ] As a screen reader user, I want to hear partial page changes or dynamic updates announced properly
+- [ ] As a keyboard user, I want focus to be maintained or reset in a predictable way when a dynamic update occurs
+
+## Environment
+
+* Operating System: all
+* Browser: all
+* Screenreading device: all
+* Server destination: all
+
+## Steps to Confirm Current Focus
+
+1. Enter [URL] in browser (Chrome works well)
+1. Right click on the page and click `Inspect` to open Chrome dev tools
+1. Click `Console` in the Dev Tools palette
+1. Type the following snippet into your console to confirm keyboard focus<br/><br/>
+    ```javascript
+    document.activeElement
+    ```
+1. Ensure focus is in line with design guidance or review with your accessibility specialist
+
+## Possible Fixes (optional)
+There are two situations where adding a `tabindex` attribute is preferable: When you need to set focus with JavaScript, or when you need to add a non-focusable element to the document focus order.
+
+### Setting focus with JavaScript
+Our rich applications often require us to update small sections of the page, or use client-side routing to change views. These are both good candidates for setting focus with JavaScript. We do this to help screen readers understand that something has changed on the page, and to avoid extra tab stops for keyboard users. Elements that will receive keyboard focus via scripting must have a `tabIndex="-1"` attribute.
+
+```html
+<h2 id="first-heading" tabIndex="-1">This heading will receive focus</h2>
+```
+
+```javascript
+/*
+ * This is a simple DOM example of setting focus through scripting.
+ * React applications will usually use the componentDidMount()
+ * lifecycle method or useEffect hooks to accomplish this task.
+ */
+const target = document.getElementById('first-heading');
+target.focus();
+```
+
+### Adding an element to the native focus order
+Sometimes it is preferable to add a non-focusable element to the native focus order without JavaScript. By adding a `tabindex="0"` to an element like a `<div>` or `<p>`, that element can receive keyboard focus when a user tabs to it. These are beneficial when we want keyboard and screen reader users to interact with things like information cards, alert boxes, or status updates.
```

Comment:
what is an example of when it is preferable?

Reply:
An example would be on the Facility Locator, when a user enters search criteria, selects Search. Focus is added to the `h2` announcing the search results. Normally, this heading would not receive focus since it isn't a focusable element, it is text.